### ISSUE TITLE
[Improvement] Have "Order by Name" also apply to leading digits in a filename (1, 2 ... 9, 10, 11 ... 19, 20, 21...)

### ISSUE THREAD

**Issue by working-name** [haiwen/seadroid#1035]
Hi there,

**Problem description:**
I'm frustrated by the app's order by name, ascending, ignoring the natural sequence of digits. I manually name some files by placing a digit at the beginning to keep the files in sequence (file names are random, can't do anything about that). However, when I look at the list, I see 1 is followed by 10, 11 and so on, before 2 come up. Then 20, 21, 22... before 3 comes up.

That is counter-intuitive and ruins intentional ordering by prepending digits.

**Expected behavior**:
I would like Order by Name to also apply to digits in a way that it maintains the natural order of numerals, where 9 is before 10, and 2 after 1.

**Other solutions explored**:
I have thought about changing my naming scheme to: a1-9, then b1-9, then c1-9... While it would be a good workaround, it then makes it difficult to ascertain how many total files there are, or how many are left to review since I go through them 1 by 1 in sequence. While a viable workaround, not any less frustrating.

Thanks for considering it.

**Comment by zhwanng** [comment]
I've noticed this problem and it's going to be optimized

**Comment by Makumasa** [comment]
It also appears that all files starting with a lowercase letter come after files starting with an uppercase letter when sorting by name. My guess is that it's simply using the ASCII numerical value when sorting?

**Comment by zhwanng** [comment]
> It also appears that all files starting with a lowercase letter come after files starting with an uppercase letter when sorting by name. My guess is that it's simply using the ASCII numerical value when sorting?
> 
> Additionally, I think sorting by name should default to "Ascending," not "Descending."

yes, the next release adds some rules, which will sort by natural numbers and characters
